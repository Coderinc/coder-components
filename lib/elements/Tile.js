import React from 'react';
import styled from 'styled-components';

const TileWrapper = styled.div`
  width: 100%;
  height: 100%;
  padding: 0.25rem;
  box-sizing: border-box;
`;

const TileContainer = styled.div`
  background: ${props =>
    props.active ? props.theme.activeBackground : props.theme.offsetBackground};
  color: ${props => props.theme.primaryFont};

  display: flex;
  align-items: center;
  height: 100%;

  box-sizing: border-box;
  border-radius: 0.125rem;
  padding: 1.5rem;

  font-size: 0.75rem;
  font-weight: lighter;
  letter-spacing: 0.0625rem;
  text-transform: ${props => (props.regularCase ? undefined : 'capitalize')};

  cursor: ${props => (props.disabled ? 'auto' : 'pointer')};
`;

const TileIcon = styled.div`
  flex: 1;
  display: flex;
  align-items: center;

  img {
    width: 2rem;
    max-height: 1.75rem;
  }
`;

const TextContainer = styled.span`
  flex: 2;
`;

export const Tile = props => (
  <TileWrapper>
    <TileContainer
      active={props.active}
      onClick={props.onClick ? () => props.onClick(props.value) : undefined}
    >
      {props.icon && (
        <TileIcon>
          <img src={props.icon} alt="Icon" />
        </TileIcon>
      )}
      <TextContainer>{props.label}</TextContainer>
    </TileContainer>
  </TileWrapper>
);

export default Tile;
