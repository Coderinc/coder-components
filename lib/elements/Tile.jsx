import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const TileWrapper = styled.div`
  width: 100%;
  height: 100%;
  padding: 0.25rem;
  box-sizing: border-box;
`;

const TileContainer = styled.div`
  background: ${({ active, theme }) => (active ? theme.activeBackground : theme.offsetBackground)};
  color: ${({ theme }) => theme.primaryFont};

  display: flex;
  align-items: center;
  height: 100%;

  box-sizing: border-box;
  border-radius: 0.125rem;
  padding: 1.5rem;

  font-size: 0.75rem;
  font-weight: lighter;
  letter-spacing: 0.0625rem;
  text-transform: ${({ regularCase }) => (regularCase ? undefined : 'capitalize')};

  cursor: ${({ disabled }) => (disabled ? 'auto' : 'pointer')};
`;

const TileIcon = styled.div`
  flex: 1;
  display: flex;
  align-items: center;

  img {
    width: 2rem;
    max-height: 1.75rem;
  }
`;

const TextContainer = styled.span`
  flex: 2;
`;

const Tile = ({
  active, onClick, value, icon, label,
}) => (
  <TileWrapper>
    <TileContainer active={active} onClick={onClick ? () => onClick(value) : undefined}>
      {icon && (
        <TileIcon>
          <img alt="Icon" src={icon} />
        </TileIcon>
      )}
      <TextContainer>
        {label}
      </TextContainer>
    </TileContainer>
  </TileWrapper>
);

Tile.propTypes = {
  active: PropTypes.bool,
  onClick: PropTypes.func,
  // eslint-disable-next-line
  value: PropTypes.any,
  icon: PropTypes.string,
  label: PropTypes.string,
};

Tile.defaultProps = {
  active: false,
  onClick: undefined,
  value: null,
  icon: null,
  label: null,
};

export default Tile;
