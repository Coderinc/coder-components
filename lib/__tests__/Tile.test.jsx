import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import sinon from 'sinon';
import 'jest-styled-components';
import Tile from '../elements/Tile';

describe('Tile', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Tile label="Test" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('fires on click', () => {
    const onClick = sinon.spy();
    const component = mount(<Tile onClick={onClick} />);
    component.find('TileContainer').simulate('click');
    expect(onClick.called).toEqual(true);
  });
})