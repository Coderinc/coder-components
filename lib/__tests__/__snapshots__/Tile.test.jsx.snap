// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  padding: 0.25rem;
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  box-sizing: border-box;
  border-radius: 0.125rem;
  padding: 1.5rem;
  font-size: 0.75rem;
  font-weight: lighter;
  -webkit-letter-spacing: 0.0625rem;
  -moz-letter-spacing: 0.0625rem;
  -ms-letter-spacing: 0.0625rem;
  letter-spacing: 0.0625rem;
  text-transform: capitalize;
  cursor: pointer;
}

.c2 {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      Test
    </span>
  </div>
</div>
`;
